### PR TITLE
253 maven build

### PR DESCRIPTION
**Pull request by kzheludkov**
Please review new maven build

### CODE REVIEW

**Comment by kzheludkov** [comment]
I will resolve conflicts...

**Comment by VladRodionov** [comment]
Ok, please resolve conflicts and run unit tests ALL to make sure nothing is broken.

**Comment by VladRodionov** [comment]
I reformatted the code according to Google as you already noticed .

File: dist/server/carrot-server-clt.sh [hunk 1]
```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
+
+START_HOME=$(dirname "$(realpath "$0")")
+echo Carrot server home directory is "${START_HOME}"
+
+cd "${START_HOME}" || exit
+
+. ./setdist.sh
+webapps="libs/${RELEASE}-${APP_PORT}"
```
Comment (VladRodionov) [inline dist/server/carrot-server-clt.sh:9]: Some artifacts of previous projects?

File: dist/server/carrot-server-clt.sh [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+start() {
+  PID=$(pid)
+  if [ ! -z "${PID}" ]; then
+    echo Server and port number in use PID="${PID}"
+    exit 1
+  fi
+
+  exec_cmd="${JAVA_HOME}/bin/java ${JVM_OPTS} org.bigbase.carrot.redis.CarrotMain ${APPS_PARAMS}"
+  echo "${exec_cmd}"
+  nohup ${exec_cmd} >>logs/catalina.log &
```
Comment (VladRodionov) [inline dist/server/carrot-server-clt.sh:37]: This is not Tomcat :)

File: dist/server/carrot-server-clt.sh [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  echo "Carrot instance ${INSTANCE_NAME} is staring on PID ${PID}, please wait..."
+
+  sleep 2
```
Comment (VladRodionov) [inline dist/server/carrot-server-clt.sh:40]: Why do we need to sleep 2 sec?

File: pom.xml [hunk 4]
```diff
@@ -1,12 +1,79 @@
...
     <groupId>org.bigbase</groupId>
     <artifactId>carrot</artifactId>
     <packaging>pom</packaging>
     <name>Carrot</name>
-    <version>1.0.0</version>
+    <version>${revision}</version>
     <url>http://www.server.com</url>
+
+    <properties>
+        <revision>1.0.0</revision>
```
Comment (VladRodionov) [inline pom.xml:14]: Set 0.3-SNAPSHOT

File: dist/server/setenv.sh [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
```
Comment (VladRodionov) [inline dist/server/setenv.sh:1]: What is this file for?

File: dist/server/setdist.sh [hunk 6]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+export APP_PORT=6379
+export JAVA_HOME=/usr/lib/jvm/java-11-openjdk-amd64
+export RELEASE=carrot-1.0.0
+export DISTRIBUTION=${RELEASE}.tar.gz
+export HEALTH_MONITOR=http://localhost:${APP_PORT}/ping
```
Comment (VladRodionov) [inline dist/server/setdist.sh:7]: We do not do HEALTH_MONITOR? Please remove

File: dist/server/carrot-server-clt.sh [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+  exit 0
+}
+
+#===== usage =====
+usage() {
+  echo
+  echo Usage:
+  # shellcheck disable=SC2102
+  echo \$\> \./carrot-server-clt.sh [start]\|[stop]\|[reboot]\|[status]
```
Comment (VladRodionov) [inline dist/server/carrot-server-clt.sh:100]: please remove status

**Review by VladRodionov — CHANGES_REQUESTED** [review]
I made some comments, please address them and remove .so file from repo. Looks very good. I am going to checkout and run some tests.

File: dist/server/carrot-server-clt.sh (hunk 1 above)
Comment (kzheludkov) [reply dist/server/carrot-server-clt.sh:9]: removed

File: dist/server/carrot-server-clt.sh (hunk 2 above)
Comment (kzheludkov) [reply dist/server/carrot-server-clt.sh:37]: renamed to carrot-stdout.log

File: dist/server/carrot-server-clt.sh (hunk 3 above)
Comment (kzheludkov) [reply dist/server/carrot-server-clt.sh:40]: it is just give a time to Mac/Linux to establish process before check if server started.
replaced to sleep 1

File: pom.xml (hunk 4 above)
Comment (kzheludkov) [reply pom.xml:14]: Done

File: dist/server/setenv.sh (hunk 5 above)
Comment (kzheludkov) [reply dist/server/setenv.sh:1]: forget to delete... 
I moved settings from setdist.sh to setenv.sh and deleted setdist.sh

File: dist/server/setdist.sh (hunk 6 above)
Comment (kzheludkov) [reply dist/server/setdist.sh:7]: removed

File: dist/server/carrot-server-clt.sh (hunk 7 above)
Comment (kzheludkov) [reply dist/server/carrot-server-clt.sh:100]: removed for now but betted to create ping available from wget to make sure server started successfully 
Will create ticket.

**Comment by VladRodionov** [comment]
Fixed some minor issues in start script, committed back to PR branch. Ran tests : all passed, but two tests took almost 45 min to finish.

**Pull request merged**